vin/app/controllers/vin/admin_controller.rb

```diff
@@ -1,47 +0,0 @@
-class Vin::AdminController < ApplicationController
-respond_to :json
-
-  # POST /admin
-  def create
-    name = params[:name]
-    admin = Admin.create({name: name})
-    if admin
-      render json: { errors: [] }
-    else
-      render json: { errors: ["Unable to create admin"]}
-    end
-  end
-
-  # GET /admin
-  def index
-    admins = Admin.all
-
-    render json: admins
-  end
-
-  # PUT /admin/{aid}
-  def update
-    admin = Admin.find(params[:id])
-    name = params[:name]
-    admin.update_attribute(:name, name)
-
-    if admin
```

Comment:
This will always be true. If there wasn't an `Admin` with that `id`, the line `Admin.find(params[:id])` will blow up. If not, `admin` will always be true. Besides, you're updating, not creating, so the error can't be `"Unable to create admin"`. People usually do this:
```ruby
admin = Admin.where(id: params[:id]).first
if admin.nil?
  render json: { errors: ["Could not find an admin with that id"] }
elsif admin.update_attributes(name: params[:name])
  render json: admin
else
  render json: { errors: admin.errors.full_messages }
end
```